core-js/src/main/javascript/cdf-legacy/components/core.js

```diff
@@ -1009,6 +1009,41 @@ var UnmanagedComponent = BaseComponent.extend({
     query.fetchData(this.parameters, handler, errorHandler);
   },
 
+  /**
+   * @summary Resolves the correct queryDefinition.
+   * @description <p>Resolves the correct queryDefinition, always defaulting to undefined.</p>
+   *
+   * @param {Object|Function} queryDef The query definition object or a getter function.
+   * @return {Object} The resolved queryDefinition object.
+   * @private
+   */
+  _resolveQueryDefinition: function(queryDef) {
+    var queryDefinition;
+    if(_.isFunction(this.getQueryDefinition)) {
+      queryDefinition = this.getQueryDefinition();
+    }
+    return this._isValidQueryDefinition(queryDefinition) ?
+        queryDefinition : (_.isFunction(queryDef) ? queryDef() : queryDef);
+  },
+
+  _isValidQueryDefinition: function (queryDef) {
+    return _.isObject(queryDef) && !_.isArray(queryDef) && !_.isEmpty(queryDef);
```

Comment:
If we are testing for an array, maybe we should also test for String and other cases were this function will return a false positive. e.g.: <code>_isValidQueryDefinition(new String("foo")) == true</code> should not be considered a valid query definition `object`.